pkg/load/load.go

```diff
@@ -0,0 +1,209 @@
+package load
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// EnvVarTag is the Go struct tag containing the environment variable that sets the option.
+	EnvVarTag = "env"
+
+	// SectionTag is the Go struct tag containing the documentation section of the option.
+	SectionTag = "sect"
+
+	// DefaultTag is the Go struct tag containing the default value of the option.
+	DefaultTag = "default"
+)
+
+var rndStringRegex = regexp.MustCompile("__RND_(\\d+)__")
+var osde2eConfig = ""
+var once sync.Once
+
+func init() {
+	rand.Seed(time.Now().Unix())
+}
+
+// IntoObject populates an object based on the tags specified in the object.
+func IntoObject(object interface{}) error {
+	// Parse out the osde2e config filename if once was provided.
+	once.Do(func() {
+		flag.StringVar(&osde2eConfig, "e2e-config", ".osde2e.yaml", "Config file for osde2e")
+		flag.Parse()
+	})
+
+	if objectType := reflect.TypeOf(object); objectType.Kind() != reflect.Ptr {
+		return fmt.Errorf("the supplied object must be a pointer")
+	}
+
+	if err := loadDefaults(object); err != nil {
+		return fmt.Errorf("error loading config defaults: %v", err)
+	}
+
+	if err := loadFromEnv(object); err != nil {
```

Comment:
This should occur after the YAML config is loaded. We want to be able to override a YAML config using Environment Variables.